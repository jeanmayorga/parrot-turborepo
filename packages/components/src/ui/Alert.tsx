import { motion } from "framer-motion";

interface Props {
  message: string;
}

export default function Alert({ message }: Props) {
  if (message === "") return null;

  return (
    <motion.div
      className="w-full opacity-0 p-0 flex items-center text-red-500 bg-red-100 rounded-lg shadow"
      animate={{ padding: "1rem", opacity: 1 }}
    >
      <div className="text-sm font-normal">{message}</div>
    </motion.div>
  );
}
