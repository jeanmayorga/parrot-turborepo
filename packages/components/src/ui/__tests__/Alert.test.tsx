import React from "react";
import renderer from "react-test-renderer";
import Alert from "../Alert";

test("Alert renders correctly", () => {
  const component = renderer.create(<Alert message="alert message" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
