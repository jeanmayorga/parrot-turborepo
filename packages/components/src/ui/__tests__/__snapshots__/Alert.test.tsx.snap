// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  className="w-full opacity-0 p-0 flex items-center text-red-500 bg-red-100 rounded-lg shadow"
  style={Object {}}
>
  <div
    className="text-sm font-normal"
  >
    alert message
  </div>
</div>
`;
